wps_jobs/wps_jobs/weather_model_jobs/ecmwf_prediction_processor.py

```diff
@@ -0,0 +1,183 @@
+
+logger = logging.getLogger(__name__)
+
+
+class ECMWFPredictionProcessor:
+    def __init__(self, stations: List[WeatherStation], model_run_repository: ModelRunRepository):        
+        self.stations = stations
+        self.model_run_repository = model_run_repository
+        self.station_predictions: Dict[int, List[ModelRunPrediction]] = defaultdict(list)
+    
+    def process(self):
+        for model_run, model in self.model_run_repository.get_prediction_model_run_timestamp_records(complete=True, interpolated=False, model_type=ModelEnum.ECMWF):
+            logger.info("model %s", model)
+            logger.info("model_run %s", model_run)
+            # Process the model run.
+            self._process_model_run(model_run)
+            # Mark the model run as interpolated.
+            self.model_run_repository.mark_model_run_interpolated(model_run)
+
+    def _process_model_run(self, model_run: PredictionModelRunTimestamp):
+        """Interpolate predictions in the provided model run for all stations."""
+        logger.info("Interpolating values for model run: %s", model_run)
+        # Iterate through stations.
+        for index, station in enumerate(self.stations):
+            logger.info("Interpolating model run %s (%s/%s) for %s:%s", model_run.id, index, len(self.stations), station.code, station.name)
+            # Process this model run for station.
+            self._process_model_run_for_station(model_run, station)
+
+    
+    def _process_model_run_for_station(self, model_run: PredictionModelRunTimestamp, station: WeatherStation):
+        """Process the model run for the provided station."""
+        # Extract the coordinate.
+        coordinate = [station.long, station.lat]
+        machine = StationMachineLearning(session=self.model_run_repository.session, model=model_run.prediction_model, target_coordinate=coordinate, station_code=station.code, max_learn_date=model_run.prediction_run_timestamp)
+        machine.learn()
+
+        # Get all the predictions associated to this particular model run.
+        model_run_predictions: List[ModelRunPrediction] = self.model_run_repository.get_model_run_predictions_for_station(station.code, model_run)
+
+        # Iterate through all the predictions.
+        prev_prediction: ModelRunPrediction | None = None
+
+        for prediction in model_run_predictions:
+            station_prediction = self._weather_station_prediction_initializer(station, model_run, prediction)
+            if prev_prediction is not None and self._should_interpolate(prev_prediction, prediction):
+                noon_prediction = construct_interpolated_noon_prediction(prev_prediction, prediction, SCALAR_MODEL_VALUE_KEYS_FOR_INTERPOLATION)
+                station_prediction = self._apply_interpolated_bias_adjustments(station_prediction, prev_prediction, noon_prediction, machine)
```

Comment:
```suggestion
                station_prediction = self._apply_interpolated_bias_adjustments(station_prediction, prev_prediction, prediction, machine)
```

Reply:
This relates to my comments on the previous thread. We have to make sure bias adjusted value interpolation happens between the original predictions (`prev_prediction` and `prediction`).

Reply:
Thank you! That makes sense, I will update this.